src/topics/resolver/global-resolver.ts

```diff
@@ -25,56 +36,69 @@ export class GlobalResolver {
+        accountSource: this.factory[regexp].accountSource,
+        accountType: this.factory[regexp].accountType,
       });
     });
   }
 
   public async resolveAll(rawData: FetchedData): Promise<ResolveAllType> {
     const updatedRawData: FetchedData = rawData;
-    const resolvedIdentifierData: FetchedData = {};
-    const accountTypes: AccountSource[] = [];
+    const accountSources: AccountSource[] = [];
+    let resolvedIdentifierData: FetchedData = {};
 
-    const resolvedFunction = async (rawDataSample: [string, BigNumberish]) => {
-      let isResolved = false;
-      for (const resolverObject of this.resolverRouter) {
-        if (
-          resolverObject.regExp &&
-          resolverObject.regExp.test(rawDataSample[0])
-        ) {
-          const resolvedAccount = await resolverObject.resolver.resolve(
-            rawDataSample[0]
-          );
-          if (resolvedAccount !== "undefined") {
-            if (!accountTypes.includes(resolverObject.accountType)) {
-              accountTypes.push(resolverObject.accountType);
-            }
+    const rawDataByAccountType: AccountsData = {};
 
-            resolvedIdentifierData[resolvedAccount] = rawDataSample[1];
-            isResolved = true;
+    let canBeResolved = false;
+    for (const [key, value] of Object.entries(rawData)) {
+      canBeResolved = false;
+      for (const resolverObject of this.resolverRouter) {
+        if (resolverObject.regExp && resolverObject.regExp.test(key)) {
+          canBeResolved = true;
+          if (!accountSources.includes(resolverObject.accountSource)) {
+            accountSources.push(resolverObject.accountSource);
+          }
+          if (
+            !Object.prototype.hasOwnProperty.call(
+              rawDataByAccountType,
+              resolverObject.accountType
+            )
+          ) {
+            rawDataByAccountType[resolverObject.accountType] = {};
           }
+          rawDataByAccountType[resolverObject.accountType][key] = value;
         }
       }
-      if (!isResolved) {
-        this.handleResolvingErrors(rawDataSample);
-        delete updatedRawData[rawDataSample[0]];
+      if (!canBeResolved && !this.ignoreAccountErrorsWhenResolving) {
+        handleResolvingErrors(`Account ${key} cannot be resolved`);
       }
-    };
+    }
 
-    await this.withConcurrency(Object.entries(rawData), resolvedFunction, {
-      concurrency: 10,
-    });
+    let resolver;
+    for (const [accountType, data] of Object.entries(rawDataByAccountType)) {
+      // find the resolver object that matches the account type
+      resolver = Object.entries(this.factory).find(
+        ([, resolverObject]) => resolverObject.accountType == accountType
+      );
+      // if resolver found, resolve the data
+      if (resolver) {
+        resolvedIdentifierData = {
+          ...resolvedIdentifierData,
+          ...(await resolver[1].resolver.resolve(data)),
+        };
+      }
```

Comment:
waht do you do if no resolver is found?

Reply:
It will just continue the loop.
And if no resolver is found at all in the object, it will return a empty object `{}`
(btw I can put te resolver var in the loop)